### PR TITLE
feat: Check pull requests using Lighthouse CI

### PR DESCRIPTION
**Pull request by smockle**
[Lighthouse CI](https://github.com/GoogleChrome/lighthouse-ci) is a CLI version of [Lighthouse](https://developers.google.com/web/tools/lighthouse), which can be used to assess site performance, accessibility, best practices and SEO.

This PR add a Lighthouse CI check to incoming pull requests, so we can automate the kinds of checks @koddsson performed [here](https://github.com/github/catalyst/pull/91#issuecomment-740645939).

We can [set thresholds](https://github.com/GoogleChrome/lighthouse-ci/blob/master/docs/configuration.md#categories) for specific categories or exclude them. [PWA](https://web.dev/progressive-web-apps/) rules are [currently excluded](https://github.com/github/catalyst/blob/e75b78a1f988c1420aef101cdd3d575cf6f7852e/lighthouserc.json#L8).

[NB](https://en.wikipedia.org/wiki/Nota_bene): As [configured](https://github.com/github/catalyst/blob/e75b78a1f988c1420aef101cdd3d575cf6f7852e/lighthouserc.json#L10-L12), results are uploaded to [temporary public storage](https://github.com/GoogleChrome/lighthouse-ci/blob/master/docs/services-disclaimer.md#temporary-public-storage). Results are not sensitive. The same results may be obtained by cloning this public repo and running `npx lhci autorun`, or by running Lighthouse via the Google Chrome DevTools against the [public site](https://github.github.io/catalyst/).

### CODE REVIEW

File: lighthouserc.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+  "ci": {
+    "collect": {
+      "url": ["http://localhost:4000/"],
```
Comment (koddsson) [inline lighthouserc.json:4]: I'd be keen on getting other URLs here in a follow up PR.

**Review by koddsson — APPROVED** [review]
This is neat! I'd love to get the issues fixed and merge this PR.

File: lighthouserc.json (hunk 1 above)
Comment (smockle) [reply lighthouserc.json:4]: Agreed! I (apparently mistakenly) thought Lighthouse CI crawled linked page, but the [`url` docs](https://github.com/GoogleChrome/lighthouse-ci/blob/master/docs/configuration.md#url) don’t mention that.

Comment (koddsson) [reply lighthouserc.json:4]: I think visits all your pages if you're just using [`staticDistDir`](https://github.com/GoogleChrome/lighthouse-ci/blob/master/docs/configuration.md#staticdistdir) and not `url`? It's not super clear to me from the documenation.

>  --autodiscoverUrlBlocklist A URL to not include when autodiscovering urls from staticDistDir.
                             Use this flag multiple times to filter multiple URLs.

**Pull request merged**